EngineeringModel/ViewModels/ElementDefinitionBrowser/ElementDefinitionsBrowserViewModel.cs

```diff
@@ -934,6 +959,46 @@ private async Task ExecuteBatchCreateSubscriptionCommand()
             }
         }
 
+        /// <summary>
+        /// Executes the <see cref="SetAsTopElementDefinitionCommand"/>
+        /// </summary>
+        private async Task ExecuteSetAsTopElementDefinitionCommand()
+        {
+            var elementDefRow = this.SelectedThing as ElementDefinitionRowViewModel;
+            if (elementDefRow?.Thing != null && (this.Thing.TopElement == null || this.Thing.TopElement != elementDefRow.Thing))
+            {
+                var iterationClone = this.Thing.Clone(false);
+                iterationClone.TopElement = elementDefRow.Thing;
+
+                var transactionContext = TransactionContextResolver.ResolveContext(this.Thing);
+                var transaction = new ThingTransaction(transactionContext);
+                transaction.CreateOrUpdate(iterationClone);
+
+                await this.DalWrite(transaction);
+            }
+        }
+
+        /// <summary>
+        /// Executes the <see cref="UnsetAsTopElementDefinitionCommandAsTopElementDefinitionCommand"/>
+        /// </summary>
+        private async Task ExecuteUnsetAsTopElementDefinitionCommand()
```

Comment:
Since this execute method is asynchronous , It's best to go async all the way => see my comments on both commands, lines [286](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pull/653/files#diff-7741444e1b5bf16cbfe77fa1f03df8cfc96db140c01c6ee715344113ae8e2fb4R286), [454](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pull/653/files#diff-7741444e1b5bf16cbfe77fa1f03df8cfc96db140c01c6ee715344113ae8e2fb4R454)